PPL command expression implementation for `geoip` (#3228)
opensearch-project/sql: docs/user/ppl/functions/geoip.rst

====================
Geo IP Address Functions
====================

.. rubric:: Table of contents

.. contents::
   :local:
   :depth: 1

> GEOIP

Author: Not really, it makes more sense to merge both .rst.

> GEOIP

Author: Actually my initial idea was that, since it's not possible to create the test  dataSource (Not dataSet) in programmatically manner during the docTest run (Unlike integration-test I can setup the dataSource on Junit).
As the result, I will need to exclude this `geoip.rst` from the DocTest, and if this got merged into `ip.rst`, the original DoctTest validation from `ip.rst` will also be skipped.

> GEOIP

Reviewer: If the test cannot be run as part of a doctest run, then exclude the `os>`.  Then the example won't be compiled and run as part of doctest. If you can add a comment in the code "why" it's being excluded, that would be nice.

> GEOIP

Author: Added comment with `os>` removed.

---------

Description
>>>>>>>>>>>

> Usage: `geoip(dataSourceName, ipAddress, options)` to lookup location information from given IP addresses via OpenSearch GeoSpatial plugin API.

Reviewer: ```suggestion
Usage: `geoip(dataSourceName, ipAddress[, options])` to lookup location information from given IP addresses via OpenSearch GeoSpatial plugin API.
```

Reviewer: It seems like the last argument is optional.

Argument type: STRING, STRING, STRING

> Return type: Tuple

Reviewer: ```suggestion
Return type: OBJECT
```

Example:

    os> source=weblogs | eval LookupResult = geoip("dataSourceName", "50.68.18.229", "country_iso_code,city_name")
    fetched rows / total rows = 1/1
    +-------------------------------------------------------------+
    | LookupResult                                                        |
    |-------------------------------------------------------------|
    | {'city_name': 'Vancouver', 'country_iso_code': 'CA'}        |
    +-------------------------------------------------------------+

Note:
 - `dataSourceName` must be an established dataSource on OpenSearch GeoSpatial plugin, detail of configuration can be found: https://opensearch.org/docs/latest/ingest-pipelines/processors/ip2geo/
 - `ip` can be an IPv4 or an IPv6 address

>  - `options` is a comma separated String for user to specify fields to output, the selection of fields subject to dataSourceProvider's schema, the list of geolite2-city dataset provide fields: "country_iso_code", "country_name", "continent_name", "region_iso_code", "region_name", "city_name", "time_zone", "location"

Reviewer: ```suggestion
 - `options` is a comma separated String of fields to output: the selection of fields is subject to dataSourceProvider's schema.  For example, the list of fields in the provided `geolite2-city` dataset includes: "country_iso_code", "country_name", "continent_name", "region_iso_code", "region_name", "city_name", "time_zone", "location"
```

Reviewer: Split that sentence, it kind of runs on...
